src/hooks/useResponsiveImages.ts

```diff
@@ -0,0 +1,13 @@
+import { useIsMediumScreen } from '@/hooks/useMaxWidth'
+import { getImageSource, type ImageObj } from '@/lib/getImageSource'
+
+const useResponsiveImages = (backgroundImage: ImageObj, image: ImageObj) => {
+  const isMediumScreen = useIsMediumScreen()
+
+  const bgImage = getImageSource(isMediumScreen, backgroundImage)
+  const imageSrc = getImageSource(isMediumScreen, image)
+
+  return { bgImage, imageSrc }
+}
```

Comment:
This is a bit confusing. What're the differences between the two? I'd suggest adding some comments.

You could make it a bit easier to follow by using arrays and then the order remains the same (and allows unlimited inputs), e.g.

```suggestion
const useResponsiveImages = (...images: Array<ImageObj>) => {
  const isMediumScreen = useIsMediumScreen()

  return images.reduce<Array<string>>((acc: Array<ImageObj>, image: ImageObj) => {
    const source = getImageSource(isMediumScreen, image)
    return acc.concat(source)
  }, [])
}
```